src/pybind/mgr/prometheus/module.py

```diff
@@ -1151,16 +1151,13 @@ def get_file_sd_config(self):
                 if service['type'] != 'mgr':
                     continue
                 id_ = service['id']
-                # get port for prometheus module at mgr with id_
-                # TODO use get_config_prefix or get_config here once
-                # https://github.com/ceph/ceph/pull/20458 is merged
                 result = CommandResult("")
                 assert isinstance(_global_instance, Module)
                 _global_instance.send_command(
                     result, "mon", '',
                     json.dumps({
-                        "prefix": "config-key get",
-                        'key': "config/mgr/mgr/prometheus/{}/server_port".format(id_),
+                        "prefix": "config get",
```

Comment:
@Rethan after a second thought, i think we should apply following patch
```diff
diff --git a/src/pybind/mgr/prometheus/module.py b/src/pybind/mgr/prometheus/module.py
index 0c2fd00d82d..2783c461b4e 100644
--- a/src/pybind/mgr/prometheus/module.py
+++ b/src/pybind/mgr/prometheus/module.py
@@ -1151,26 +1151,8 @@ class Module(MgrModule):
                 if service['type'] != 'mgr':
                     continue
                 id_ = service['id']
-                # get port for prometheus module at mgr with id_
-                # TODO use get_config_prefix or get_config here once
-                # https://github.com/ceph/ceph/pull/20458 is merged
-                result = CommandResult("")
-                assert isinstance(_global_instance, Module)
-                _global_instance.send_command(
-                    result, "mon", '',
-                    json.dumps({
-                        "prefix": "config-key get",
-                        'key': "config/mgr/mgr/prometheus/{}/server_port".format(id_),
-                    }),
-                    "")
-                r, outb, outs = result.wait()
-                if r != 0:
-                    _global_instance.log.error("Failed to retrieve port for mgr {}: {}".format(id_, outs))
-                    targets.append('{}:{}'.format(hostname, DEFAULT_PORT))
-                else:
-                    port = json.loads(outb)
-                    targets.append('{}:{}'.format(hostname, port))
-
+                port = self._get_module_option('server_port', DEFAULT_PORT, id_)
+                targets.append(f'{hostname}:{port}')
         ret = [
             {
                 "targets": targets,
```

as i think what @jan--f wanted is to a higher level method instead of using `send_command()`.

Reply:
That does make codes a lot simpler ;-). However I was wondering whether we want to call a method with prefix "_" （looks like an internal method), but it doesn't bother me and it does pass the self-test.

Reply:
i included the proposed change in #38264 for ease of testing, will drop it once you are fine with it and incorporate it.

Reply:
That's okay, so this PR can be closed.

Reply:
it's indeed an internal method. but probably we can look at it in a different way -- it's internal because we don't expect to read settings of other mgr instance. but in this very use case, we enumerate settings of **all** mgr instances.

[off-topic] actually, i think this is another form of single point of failure. as we should always get the contact point of mgr from monitor.

anyway, i still believe this is the way to go.